### ISSUE TITLE
Incompatible return value type `type[Enum]` when returning a functional-style Enum

### ISSUE THREAD

**Issue by stefanhoelzl** [python/mypy#17147]
**Bug Report**
When returning a functional-style `Enum` class from a function with the return type `type[Enum]` mypy reports an error.

When the created `Enum` class is assigned to a variable and the variable is return, mypy does not report an error.

**To Reproduce**

```python
from enum import Enum

def direct_return() -> type[Enum]:
    return Enum("direct", {"a": 0})    # error: Incompatible return value type (got "Enum", expected "type[Enum]")  [return-value]

def indirect_return() -> type[Enum]:
    indirect = Enum("indirect", {"a": 0})
    return indirect                    # no error as expected
```

[Gist URL](https://gist.github.com/mypy-play/9d85f8dda9329726c5ce6397904c4ba1)
[Playground URL](https://mypy-play.net/?mypy=master&python=3.12&gist=9d85f8dda9329726c5ce6397904c4ba1)

**Expected Behavior**

no mypy warnings

**Actual Behavior**

main.py:4: error: Incompatible return value type (got "Enum", expected "type[Enum]")  [return-value]

**Your Environment**

- Mypy version used: `1.9.0` and `master` on [mypy-play.net](https://mypy-play.net)
- Mypy command-line flags: [mypy-play.net](https://mypy-play.net) defaults
- Mypy configuration options from `mypy.ini` (and other config files): [mypy-play.net](https://mypy-play.net) defaults
- Python version used: `3.12`
